pandasai/helpers/anonymizer.py

```diff
@@ -175,5 +178,12 @@ def anonymize_dataframe_head(
                 pd.eval(cell_value) if cell_value in ["True", "False"] else cell_value
             )
     # restore the original data types
-    data_frame = data_frame.astype(dtypes)
+    for i in range(len(data_frame.columns)):
```

Comment:
Also, do you think we can have a unit test for this?

Reply:
Added one unit test